### DISCUSSION TITLE
developがデフォルトブランチであるべきでは？

### DISCUSSION THREAD

**Discussion by h-takesg** [kusumotolab/kJuiceApp#77] [General]
# 現在
`main` デフォルトブランチ，releaseとしての役割
`develop`開発ブランチ，ここからブランチを生やす

# GitHubの想定？
`main`開発ブランチ，開発者間で共有される最新版
リリースブランチなし
tagで管理

# misc
Issueの自動リンク自動クローズはデフォルトブランチに対してのPRのときだけらしい
[ref](https://docs.github.com/ja/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#:~:text=%E6%B3%A8%E9%87%88%3A%20pull%20request%20%E3%81%AB%E3%81%8A%E3%81%91%E3%82%8B%E7%89%B9%E5%88%A5%E3%81%AA%E3%82%AD%E3%83%BC%E3%83%AF%E3%83%BC%E3%83%89%E3%81%AF%E3%80%81pull%20request%20%E3%81%8C%E3%83%AA%E3%83%9D%E3%82%B8%E3%83%88%E3%83%AA%E3%81%AE%E3%83%87%E3%83%95%E3%82%A9%E3%83%AB%E3%83%88%E3%83%96%E3%83%A9%E3%83%B3%E3%83%81%E3%82%92%E3%82%BF%E3%83%BC%E3%82%B2%E3%83%83%E3%83%88%E3%81%99%E3%82%8B%E3%81%A8%E3%81%8D%E3%81%AB%E8%A7%A3%E9%87%88%E3%81%95%E3%82%8C%E3%81%BE%E3%81%99%E3%80%82%20%E3%81%9F%E3%81%A0%E3%81%97%E3%80%81PR%20%E3%81%AE%E3%83%99%E3%83%BC%E3%82%B9%E3%81%8C%E3%81%9D%E3%82%8C%E4%BB%A5%E5%A4%96%E3%81%AE%E3%83%96%E3%83%A9%E3%83%B3%E3%83%81%E3%81%A7%E3%81%82%E3%82%8B%E5%A0%B4%E5%90%88%E3%80%81%E3%81%9D%E3%82%8C%E3%82%89%E3%81%AE%E3%82%AD%E3%83%BC%E3%83%AF%E3%83%BC%E3%83%89%E3%81%AF%E7%84%A1%E8%A6%96%E3%81%95%E3%82%8C%E3%80%81%E3%83%AA%E3%83%B3%E3%82%AF%E3%81%AF%E4%BD%9C%E6%88%90%E3%81%95%E3%82%8C%E3%81%BE%E3%81%9B%E3%82%93%E3%80%82PR%20%E3%81%AE%E3%83%9E%E3%83%BC%E3%82%B8%E3%81%AF%E3%81%93%E3%81%AE%20issue%20%E3%81%AB%E5%AF%BE%E3%81%97%E3%81%A6%E4%BD%95%E3%81%AE%E5%8A%B9%E6%9E%9C%E3%82%82%E6%8C%81%E3%81%A1%E3%81%BE%E3%81%9B%E3%82%93%E3%80%82%20%E3%82%AD%E3%83%BC%E3%83%AF%E3%83%BC%E3%83%89%E3%81%AE%201%20%E3%81%A4%E3%82%92%E4%BD%BF%E3%81%A3%E3%81%A6%20pull%20request%20%E3%82%92%20issue%20%E3%81%AB%E3%83%AA%E3%83%B3%E3%82%AF%E3%81%99%E3%82%8B%E5%A0%B4%E5%90%88%E3%80%81PR%20%E3%81%8C%E6%97%A2%E5%AE%9A%E3%83%96%E3%83%A9%E3%83%B3%E3%83%81%E4%B8%8A%E3%81%AB%E3%81%AA%E3%81%91%E3%82%8C%E3%81%B0%E3%81%AA%E3%82%8A%E3%81%BE%E3%81%9B%E3%82%93%E3%80%82)

**Comment by grtw2116** [comment]
mainをリリース管理にしか使わないのであれば，ひとまずv1.0.0リリースまではdevelopをデフォルトにしたほうが開発上の恩恵を多く受けられると思います．
v1.0.0リリース後は，どちらのブランチをデフォルトにするかリリース間隔で様子見するのが良さそうです．

**Comment by h-takesg** [comment]
「現行developをデフォルトにするだけでいいのではないか」
確かに同意
付け加えて，リリース管理ブランチが不要ではって気がしてます

* developをデフォルトブランチに変更
* mainを削除
* リリース管理はタグを使用

がいいかなと思ってる

**Comment by grtw2116** [comment]
デフォルトブランチを変更しました．
- `main`を`main-old`にリネーム
- `develop`を`main`にリネームしデフォルトブランチに変更

**Comment by grtw2116** [comment]
`main-old`は一応残してるだけなので，作業に支障がなければ削除してもらってもOKです
